src/service/service_group_access.go

```diff
@@ -0,0 +1,84 @@
+package service
+
+import (
+	"api5back/ent"
+	"api5back/ent/department"
+	"api5back/src/model"
+	"context"
+	"errors"
+)
+
+type GroupAcessReturn struct {
+	Id          int                `json:"id"`
+	Name        string             `json:"name"`
+	Departments []model.Suggestion `json:"departments"`
+}
+
+type CreateGroupAcessRequest struct {
+	Name          string `json:"name" binding:"required"`
+	DepartmentIDs []int  `json:"departments" binding:"required"`
+}
+
+// GetGroupAcessWithDepartments retorna grupos de acesso com seus departamentos
+func GetGroupAcessWithDepartments(
+	ctx context.Context,
+	client *ent.Client,
+) ([]GroupAcessReturn, error) {
+	// Consulta todos os grupos de acesso e os departamentos relacionados
+	groups, err := client.GroupAcess.Query().
+		WithDepartment(). // Inclui os departamentos relacionados
+		All(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// Mapeia o resultado para o formato esperado
+	var response []GroupAcessReturn
+	for _, group := range groups {
+		var departments []model.Suggestion
+		for _, dept := range group.Edges.Department {
```

Comment:
considerar a possibilidade de erro ao tentar pegar os `Department` do `GroupAccess`:

```suggestion
		groupAccessDepartments, err := group.Edges.DepartmentOrErr()
		if err != nil {
		    return nil, fmt.Errorf("failed to retrieve `Department` of `GroupAccess: %w`, err)
		}
		
		for _, dept := range groupAccessDepartments {
```